angular/playground/projects/ng-entity-service/src/lib/ng-entity-service.spec.ts

```diff
@@ -0,0 +1,1435 @@
+      (service: TestServiceWithInlineConfig, httpMock: HttpTestingController) => {
+        const dummyEntity: TestEntity = { id: 1, foo: 'foo', bar: 123 };
+        service.add(dummyEntity, { params: { foo: 'foo', bar: 'bar' } }).subscribe();
+
+        const req = httpMock.expectOne(x => x.url === service.api);
+        expect(req.request.params.get('foo')).toEqual('foo');
+        expect(req.request.params.get('bar')).toEqual('bar');
+        req.flush([]);
+      }
+    ));
+
+    it('should dispatch success notice when request is successful', inject(
+      [TestServiceWithInlineConfig, HttpTestingController, NgEntityServiceNotifier],
+      (service: TestServiceWithInlineConfig, httpMock: HttpTestingController, notifier: NgEntityServiceNotifier) => {
+        spyOn(notifier, 'dispatch');
+        const dummyEntity: TestEntity = { id: 1, foo: 'foo', bar: 123 };
+        service.add(dummyEntity).subscribe(() => {
+          expect(notifier.dispatch).toHaveBeenCalledWith({
+            storeName,
+            type: 'success',
+            payload: dummyEntity,
+            method: HttpMethod.POST,
+            successMsg: undefined
+          });
+        });
+
+        const req = httpMock.expectOne(service.api);
+        req.flush(dummyEntity);
+      }
+    ));
+
+    it('should dispatch success notice with configured success message when successMsg is configured and request is successful', inject(
+      [TestServiceWithInlineConfig, HttpTestingController, NgEntityServiceNotifier],
+      (service: TestServiceWithInlineConfig, httpMock: HttpTestingController, notifier: NgEntityServiceNotifier) => {
+        spyOn(notifier, 'dispatch');
+        const successMsg = 'Success test message...';
+        const dummyEntity: TestEntity = { id: 1, foo: 'foo', bar: 123 };
+        service.add(dummyEntity, { successMsg }).subscribe(() => {
+          expect(notifier.dispatch).toHaveBeenCalledWith({
+            storeName,
+            type: 'success',
+            payload: dummyEntity,
+            method: HttpMethod.POST,
+            successMsg
+          });
+        });
+
+        const req = httpMock.expectOne(service.api);
+        req.flush(dummyEntity);
+      }
+    ));
+
+    it('should dispatch error notice when request gives an error', inject(
+      [TestServiceWithInlineConfig, HttpTestingController, NgEntityServiceNotifier],
+      (service: TestServiceWithInlineConfig, httpMock: HttpTestingController, notifier: NgEntityServiceNotifier) => {
+        spyOn(notifier, 'dispatch');
+        const dummyEntity: TestEntity = { id: 1, foo: 'foo', bar: 123 };
+        service.add(dummyEntity).subscribe({
+          error: () =>
+            expect(notifier.dispatch).toHaveBeenCalledWith(
+              jasmine.objectContaining({
+                storeName,
+                type: 'error',
+                method: HttpMethod.POST,
+                errorMsg: '' // !WARN this is inconsistant with the successMsg, as that is undefined
```

Comment:
I'm not sure I understood the difference, can you explain, please?

Reply:
the successMsg is `successMsg: undefined` compared to `errorMsg: ''` when not set.

This is because handleError is defined like this `private handleError(method: HttpMethod, error: any, errorMsg: string)` (unchanged). errorMsg isn't optional so I had to call it with `''`. It didn't before but the code, but the code was making an illegal call.

Now that I think about it, it probably was `errorMsg: undefined` before.

Guess I should try to run the tests against the old code

Reply:
confirmed! it was `errorMsg: undefined` 

I'll fix